src/openrct2/peep/Guest.cpp

```diff
@@ -1591,9 +1569,9 @@ bool Guest::DecideAndBuyItem(Ride* ride, int32_t shopItem, money32 price)
                 if (itemValue > (static_cast<money16>(scenario_rand() & 0x07)))
                 {
                     // "I'm not paying that much for x"
-                    PeepThoughtType thought_type = static_cast<PeepThoughtType>(
-                        (shopItem >= 32 ? (PEEP_THOUGHT_TYPE_PHOTO2_MUCH + (shopItem - 32))
-                                        : (PEEP_THOUGHT_TYPE_BALLOON_MUCH + shopItem)));
+                    PeepThoughtType thought_type = static_cast<PeepThoughtType>((
+                        shopItem >= ShopItem::Photo2 ? (ShopItem(PEEP_THOUGHT_TYPE_PHOTO2_MUCH) + (shopItem - ShopItem::Photo2))
```

Comment:
This is wrong. It should be:
```
PeepThoughtType thought_type = 
                        shopItem >= ShopItem::Photo2 ? PEEP_THOUGHT_TYPE_PHOTO2_MUCH + static_cast<uint8_t>((shopItem - ShopItem::Photo2))) 
                                                     : PEEP_THOUGHT_TYPE_BALLOON_MUCH + static_cast<uint8_t>(shopItem);
```

Reply:
I have corrected it, but `static_cast<PeepThoughtType>` had to be left to avoid compilation errors.